### DISCUSSION TITLE
Is XmnlsDefinition supported for local Maui app assembly?

### DISCUSSION THREAD

**Discussion by Axemasta** [dotnet/maui#18016] [Q&A]
The maui app i'm working on is starting to get a little view heavy and I have organized my viewmodels into folders, meaning they occupy different namespaces. Something I've usually done for Xamarin apps is have a seperate assembly for things like controls, effects and then reference them with an `XmnlsDefinition` in my xaml since its super clean.

I was playing around with my viewmodel namespace seeing if I could do the same, my viewmodels all live within my Maui app assembly:
```csharp
[assembly: XmlnsDefinition("http://schemas.mycompany.com/2023/myapp/viewmodels", "MyApp.ViewModels")]
[assembly: XmlnsDefinition("http://schemas.mycompany.com/2023/myapp/viewmodels", "MyApp.ViewModels.SubDirectoryOne")]
[assembly: XmlnsDefinition("http://schemas.mycompany.com/2023/myapp/viewmodels", "MyApp.ViewModels.SubDirectoryTwo")]
```

This works with xaml intellisense in Rider, when specifying the compiled bindings for my view:
```xml
<ContentPage 
    xmlns:viewmodels="http://schemas.mycompany.com/2023/myapp/viewmodels"
    x:DataType="viewmodels:HomeViewModel" />
```

However when compiling I do see an error:
```
2>HomePage.xaml(66,7): 
Error XFC0000 XamlC: Cannot resolve type "http://schemas.mycompany.com/2023/myapp/viewmodels:viewmodels:HomeViewModel".
```

I have implemented `XmlnsDefinition`s before, I have 2 in my app (in 2 seperate class libraries) and multiple in custom libraries I am consuming for my app. I have seen the info from Stephane [here for maui](https://github.com/dotnet/maui/issues/8158#issuecomment-1160118898).

The attributes are setup correctly, I have hard references to my views and viewmodels (I am using Prism and have registered them, manually creating instances of my viewmodels still doesnt fix this).

Is this just not supported? Should this be supported (It would certainly be helpful!). If its not supported can we update the documentation to make this clear. I found an [SO post for WPF](https://stackoverflow.com/a/2778489/8828057) saying it's not supported but that was 13 years ago... Are things still this way? 🙄

**Comment by jmichas** [comment]
@Axemasta Did you ever solve this? I am facing the same problem, but it only occurs when consuming the Xmlns in a resource dictionary file that gets merged.
